### ISSUE TITLE
pyplot.bar does not cycle through colors (>2.0.0b1)

### ISSUE THREAD

**Issue by msspektor** [matplotlib/matplotlib#7891]
I hope I'm not reporting an issue that's already known. Starting with matplotlib==2.0.0b1, pyplot.bar does not cycle through colors anymore. Instead, if I pass list of colors, it only uses color 'C0'.

Minimum code to reproduce bug:

```python
import matplotlib.pyplot as plt
x_pos = [0, 1, 2]
heights = [1, 2, 3]
colors = ['0.2', '0.5', '0.8']
print('Works properly:')
[plt.bar(left=x_pos[ii], height=heights[ii], color=colors[ii]) for ii in range(len(x_pos))]
plt.show()
print('Does not work aynmore:')
plt.bar(left=x_pos, height=heights, color=colors)
plt.show()
```
Output:

<img width="382" alt="img" src="https://cloud.githubusercontent.com/assets/20988354/22149638/2e2c2d96-df15-11e6-8759-8e95506a2ac8.png">

Here's the output using the same code with matplotlib==1.5.3

<img width="446" alt="img2" src="https://cloud.githubusercontent.com/assets/20988354/22149823/42868696-df16-11e6-9641-57a29968a766.png">

**Comment by tacaswell** [comment]
Thanks for reporting this.  I do not think this is a duplicate, but falls into a common class of problem where there are some corner cases in separating a single rgb(a) value from a sequence of colors.

That blue isn't actually 'C0', just very close.  What is happening here is that your 3 gray scale values are being treated as a single rgb value, that ends up being _close_ to 'C0', ex

```python
import matplotlib.pyplot as plt
import matplotlib.colors as mcolors
plt.ion()
x_pos = [0, 1, 2]
heights = [1, 2, 3]
colors = ['0.2', '0.5', '0.8']
colors_rgb = [mcolors.to_rgb(c) for c in ['0.2', '0.5', '0.8']]
fig, (ax1, ax2, ax3) = plt.subplots(3, 1)

ax1.bar(left=x_pos, height=heights, color=colors)
ax2.bar(left=x_pos, height=heights, color=colors[::-1])
ax3.bar(left=x_pos, height=heights, color=colors_rgb)
for a in (ax1, ax2, ax3):
    a.axhline(.2, lw=5, color='C0')
```

![so](https://cloud.githubusercontent.com/assets/199813/22153627/11c48ea6-def6-11e6-8dc5-b6e8e53f3686.png)


attn @anntzer I think this is a bug in `to_rgba_array`:

```python
In [24]: mcolors.to_rgba_array(['0.2', '0.5', '0.8'])
Out[24]: 
array([[ 0.2,  0.5,  0.8,  1. ]])

In [25]: [mcolors.to_rgb(c) for c in ['0.2', '0.5', '0.8']]
Out[25]: 
[(0.2, 0.2, 0.2), (0.5, 0.5, 0.5), (0.8, 0.8, 0.8)]

```

**Comment by anntzer** [comment]
See #7896.

**Comment by NelleV** [comment]
closed by #7896 7896
